cvat-ui/src/actions/annotation-actions.ts

```diff
@@ -295,28 +295,59 @@ async function fetchAnnotations(frameNumber?: number): Promise<{
     };
 }
 
+function navigationInterruptionDetected(callback: (interrupted: boolean) => void): void {
+    let currentPath = window.location.pathname;
+
+    // Listen for popstate events (back/forward navigation)
+    window.addEventListener('popstate', () => {
+        const newPath = window.location.pathname;
+        if (newPath !== currentPath) {
+            console.log('Navigation interruption detected.');
+            currentPath = newPath;
+
+            // Notify the callback that navigation interruption occurred
+            callback(true);
+        }
+    });
+}
 export function fetchAnnotationsAsync(): ThunkAction {
     return async (dispatch: ActionCreator<Dispatch>): Promise<void> => {
         try {
             const {
                 states, history, minZ, maxZ,
             } = await fetchAnnotations();
 
-            dispatch({
-                type: AnnotationActionTypes.FETCH_ANNOTATIONS_SUCCESS,
-                payload: {
-                    states,
-                    history,
-                    minZ,
-                    maxZ,
-                },
+            // Check if navigation interruption occurred
+            navigationInterruptionDetected((interrupted: boolean) => {
```

Comment:
From the code I see:

1. We fetch annotations in line 318
2. We call `navigationInterruptionDetected` with callback passed
2.1. Callback will be called only if `postate` event was dispatched with `true` argument.
2.2. Callback never called with `false` argument. 

Condition in line 322 is always false, so, I am not sure how this code is supposed to work. Did you test it?

Reply:
Ok, I also see errors in my logic of handling this now. What do you think of handling this issue in such away.

1. I separate the handling of navigation interruptions from the handling of fetching errors.
2. I use a flag to track whether a navigation interruption occurred during the fetching process.
3. Then check if navigation interruption occurred , dispatch success action if no interruption AND dispatch FETCH_ANNOTATIONS_FAILED only if not due to navigation interruption
4. Then navigationInterruptionDetected function now returns a function that removes the event listener when it's no longer needed. So that that the event listener is cleaned up to prevent memory leaks.